drivers/display/ssd1673.c

```diff
@@ -0,0 +1,532 @@
+	tmp[1] = SSD1673_VAL_GDV_CTRL_B;
+	if (ssd1673_write_cmd(driver, SSD1673_CMD_GDV_CTRL, tmp, 2)) {
+		return -1;
+	}
+
+	tmp[0] = SSD1673_VAL_SDV_CTRL;
+	if (ssd1673_write_cmd(driver, SSD1673_CMD_SDV_CTRL, tmp, 1)) {
+		return -1;
+	}
+
+	tmp[0] = SSD1673_VAL_VCOM_VOLTAGE;
+	if (ssd1673_write_cmd(driver, SSD1673_CMD_VCOM_VOLTAGE, tmp, 1)) {
+		return -1;
+	}
+
+	tmp[0] = SSD1673_VAL_DUMMY_LINE;
+	if (ssd1673_write_cmd(driver, SSD1673_CMD_DUMMY_LINE, tmp, 1)) {
+		return -1;
+	}
+
+	tmp[0] = SSD1673_VAL_GATE_LWIDTH;
+	if (ssd1673_write_cmd(driver, SSD1673_CMD_GATE_LINE_WIDTH, tmp, 1)) {
+		return -1;
+	}
+
+	ssd1673_set_orientation(driver);
+	driver->numof_part_cycles = 0;
+	driver->last_lut = SSD1673_LAST_LUT_INITIAL;
+	driver->contrast = 0;
+
+	return 0;
+}
+
+static int ssd1673_init(struct device *dev)
+{
+	struct ssd1673_data *driver = dev->driver_data;
+
+	LOG_DBG("");
+
+	driver->spi_dev = device_get_binding(CONFIG_SSD1673_SPI_DEV_NAME);
+	if (driver->spi_dev == NULL) {
+		LOG_ERR("Could not get SPI device for SSD1673");
+		return -EIO;
+	}
+
+	driver->spi_config.frequency = CONFIG_SSD1673_SPI_FREQ;
+	driver->spi_config.operation = SPI_OP_MODE_MASTER | SPI_WORD_SET(8);
+	driver->spi_config.slave = CONFIG_SSD1673_SPI_SLAVE_NUMBER;
+	driver->spi_config.cs = NULL;
+
+	driver->reset = device_get_binding(CONFIG_SSD1673_RESET_GPIO_PORT_NAME);
+	if (driver->reset == NULL) {
+		LOG_ERR("Could not get GPIO port for SSD1673 reset");
+		return -EIO;
+	}
+
+	gpio_pin_configure(driver->reset, CONFIG_SSD1673_RESET_PIN,
+			   GPIO_DIR_OUT);
+
+	driver->dc = device_get_binding(CONFIG_SSD1673_DC_GPIO_PORT_NAME);
+	if (driver->dc == NULL) {
+		LOG_ERR("Could not get GPIO port for SSD1673 DC signal");
+		return -EIO;
+	}
+
+	gpio_pin_configure(driver->dc, CONFIG_SSD1673_DC_PIN,
+			   GPIO_DIR_OUT);
+
+	driver->busy = device_get_binding(CONFIG_SSD1673_BUSY_GPIO_PORT_NAME);
+	if (driver->busy == NULL) {
+		LOG_ERR("Could not get GPIO port for SSD1673 busy signal");
+		return -EIO;
+	}
+
+	gpio_pin_configure(driver->busy, CONFIG_SSD1673_BUSY_PIN,
+			   GPIO_DIR_IN);
+
+#if defined(CONFIG_SSD1673_SPI_GPIO_CS)
+	driver->cs_ctrl.gpio_dev = device_get_binding(
+		CONFIG_SSD1673_SPI_GPIO_CS_DRV_NAME);
+	if (!driver->cs_ctrl.gpio_dev) {
+		LOG_ERR("Unable to get SPI GPIO CS device");
+		return -EIO;
+	}
+
+	driver->cs_ctrl.gpio_pin = CONFIG_SSD1673_SPI_GPIO_CS_PIN;
+	driver->cs_ctrl.delay = 0;
+	driver->spi_config.cs = &driver->cs_ctrl;
+#endif
+
+	ssd1673_controller_init(dev);
+
+	return 0;
+}
+
+static struct ssd1673_data ssd1673_driver;
+
+static struct display_driver_api ssd1673_driver_api = {
+	.blanking_on = ssd1673_resume,
+	.blanking_off = ssd1673_suspend,
+	.write = ssd1673_write,
+	.read = ssd1673_read,
+	.get_framebuffer = ssd1673_get_framebuffer,
+	.set_brightness = ssd1673_set_brightness,
+	.set_contrast = ssd1673_set_contrast,
+	.get_capabilities = ssd1673_get_capabilities,
+	.set_pixel_format = ssd1673_set_pixel_format,
+	.set_orientation = NULL,
```

Comment:
API expects set_orientation to be implemented, no null checking is in place at API level